scripts/installer_test.go

```diff
@@ -0,0 +1,339 @@
+)
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+	if *msiPathFlag == "" {
+		log.Fatal("no msi passed, use `go test ... -args -msi \"<msi path (using \\ seperator)>\"`")
+	}
+
+	tmp, err := ioutil.TempDir("", "release")
+	if err != nil {
+		panic(err)
+	}
+
+	releaseMSIPath = filepath.Join(tmp, oldRelease+".msi")
+	if err = downloadInstaller(oldRelease, releaseMSIPath); err != nil {
+		panic(err)
+	}
+	defer func() {
+		err = errs.Combine(os.RemoveAll(tmp))
+	}()
+
+	os.Exit(m.Run())
+}
+
+func TestInstaller_Config(t *testing.T) {
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	installDir := ctx.Dir("install")
+	configPath := ctx.File("install", "config.yaml")
+
+	walletAddr := "0x0000000000000000000000000000000000000000"
+	email := "user@mail.test"
+	publicAddr := "127.0.0.1:10000"
+
+	args := []string{
+		fmt.Sprintf("STORJ_WALLET=%s", walletAddr),
+		fmt.Sprintf("STORJ_EMAIL=%s", email),
+		fmt.Sprintf("STORJ_PUBLIC_ADDRESS=%s", publicAddr),
+	}
+	install(t, ctx, *msiPathFlag, installDir, args...)
+	defer requireUninstall(t, ctx)
+
+	expectedEmail := fmt.Sprintf("operator.email: %s", email)
+	expectedWallet := fmt.Sprintf("operator.wallet: \"%s\"", walletAddr)
+	expectedAddr := fmt.Sprintf("contact.external-address: %s", publicAddr)
+
+	configStr := readConfigLines(t, ctx, configPath)
+
+	require.Contains(t, configStr, expectedEmail)
+	require.Contains(t, configStr, expectedWallet)
+	require.Contains(t, configStr, expectedAddr)
+}
+
+func TestUpgrade_Config(t *testing.T) {
+	if *msiPathFlag == "" {
+		t.Fatal("no msi passed, use `go test ... -args -msi <msi path>`")
+	}
+
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	installDir := ctx.Dir("install")
+	configPath := ctx.File("install", "config.yaml")
+
+	walletAddr := "0x0000000000000000000000000000000000000000"
+	email := "user@mail.test"
+	publicAddr := "127.0.0.1:10000"
+
+	// install old release
+	args := []string{
+		fmt.Sprintf("STORJ_WALLET=%s", walletAddr),
+		fmt.Sprintf("STORJ_EMAIL=%s", email),
+		fmt.Sprintf("STORJ_PUBLIC_ADDRESS=%s", publicAddr),
+	}
+	install(t, ctx, releaseMSIPath, installDir, args...)
+
+	// upgrade using test msi
+	install(t, ctx, *msiPathFlag, installDir)
+	defer requireUninstall(t, ctx)
+
+	expectedEmail := fmt.Sprintf("operator.email: %s", email)
+	expectedWallet := fmt.Sprintf("operator.wallet: \"%s\"", walletAddr)
+	expectedAddr := fmt.Sprintf("contact.external-address: %s", publicAddr)
+
+	configStr := readConfigLines(t, ctx, configPath)
+
+	require.Contains(t, configStr, expectedEmail)
+	require.Contains(t, configStr, expectedWallet)
+	require.Contains(t, configStr, expectedAddr)
+}
+
+func install(t *testing.T, ctx *testcontext.Context, msiPath, installDir string, args ...string) {
+	t.Logf("installing from %s\n", msiPath)
+	logPath := ctx.File("log", "install.log")
+	args = append(append([]string{
+		"/i", msiPath,
+		"/log", logPath,
+	}, append(msiBaseArgs, "INSTALLFOLDER="+installDir)...), args...)
```

Comment:
`INSTALLFOLDER` should not be set for upgrades. The installer should figure it out on its own when in upgrade mode.

Reply:
Good point. Will fix